Compilation error with Arduino Core 3.0.0 (esp-idf-v5.1-libs branch)

### Board

ESP32-C3

### Device Description

C3 Devboard

### Hardware Configuration

N/A

### Version

other

### IDE Name

Arduino IDE

### Operating System

Win11

### Flash frequency

40MHz

### PSRAM enabled

no

### Upload speed

115200

### Description

When compiling any sketch with the ESP32-C3 and Debug Level as Verbose, a compilation error occurs.

### Sketch

```cpp
// Any sketch, including a BareMinimum example like this

void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}
```


### Debug Message

```plain
<command-line>: warning: ISO C99 requires whitespace after the macro name
In file included from C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal.h:76,
                 from C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal-cpu.c:25:
C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal-cpu.c: In function 'setCpuFrequencyMhz':
C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal-cpu.c:244:108: error: 'RTC_CPU_FREQ_SRC_APLL' undeclared (first use in this function); did you mean 'RTC_CPU_FREQ_SRC_PLL'?
  244 |     log_d("%s: %u / %u = %u Mhz, APB: %u Hz", (conf.source == RTC_CPU_FREQ_SRC_PLL)?"PLL":((conf.source == RTC_CPU_FREQ_SRC_APLL)?"APLL":((conf.source == RTC_CPU_FREQ_SRC_XTAL)?"XTAL":"8M")), conf.source_freq_mhz, conf.div, conf.freq_mhz, apb);
      |                                                                                                            ^~~~~~~~~~~~~~~~~~~~~
C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal-log.h:111:72: note: in definition of macro 'log_d'
  111 | #define log_d(format, ...) log_printf(ARDUHAL_LOG_FORMAT(D, format), ##__VA_ARGS__)
      |                                                                        ^~~~~~~~~~~
C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal-cpu.c:244:108: note: each undeclared identifier is reported only once for each function it appears in
  244 |     log_d("%s: %u / %u = %u Mhz, APB: %u Hz", (conf.source == RTC_CPU_FREQ_SRC_PLL)?"PLL":((conf.source == RTC_CPU_FREQ_SRC_APLL)?"APLL":((conf.source == RTC_CPU_FREQ_SRC_XTAL)?"XTAL":"8M")), conf.source_freq_mhz, conf.div, conf.freq_mhz, apb);
      |                                                                                                            ^~~~~~~~~~~~~~~~~~~~~
C:\Users\rocor\Documents\Arduino\hardware\espressif\esp32-SuGlider\cores\esp32\esp32-hal-log.h:111:72: note: in definition of macro 'log_d'
  111 | #define log_d(format, ...) log_printf(ARDUHAL_LOG_FORMAT(D, format), ##__VA_ARGS__)
      |                                                                        ^~~~~~~~~~~
```


### Other Steps to Reproduce

No need for flasing. It is a compilation error.

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

--- a/cores/esp32/esp32-hal-cpu.c
+++ b/cores/esp32/esp32-hal-cpu.c
@@ -241,7 +241,11 @@ bool setCpuFrequencyMhz(uint32_t cpu_freq_mhz){
     if(apb_change_callbacks){
         triggerApbChangeCallback(APB_AFTER_CHANGE, capb, apb);
     }
+#ifdef SOC_CLK_APLL_SUPPORTED
     log_d("%s: %u / %u = %u Mhz, APB: %u Hz", (conf.source == RTC_CPU_FREQ_SRC_PLL)?"PLL":((conf.source == RTC_CPU_FREQ_SRC_APLL)?"APLL":((conf.source == RTC_CPU_FREQ_SRC_XTAL)?"XTAL":"8M")), conf.source_freq_mhz, conf.div, conf.freq_mhz, apb);
+#else
+    log_d("%s: %u / %u = %u Mhz, APB: %u Hz", (conf.source == RTC_CPU_FREQ_SRC_PLL)?"PLL":((conf.source == RTC_CPU_FREQ_SRC_XTAL)?"XTAL":"17.5M"), conf.source_freq_mhz, conf.div, conf.freq_mhz, apb);
+#endif
     return true;
 }
 

